File: day1/part2.py
input = open("input.txt","r")
inpArr = map(int, input.read().splitlines())
total = 0
i = 0
totals = set()

while total not in totals:
    totals.add(total)
    total += inpArr[i % len(inpArr)]
    i+=1

print("Day 1 | Part 2 - Solution: {}".format(total))

File: day1/part2.js
const { readFileSync } = require("fs");
const inpArr = readFileSync("input.txt").toString().split("\n");
const totals = new Set();
let total = 0;
let i = 0;

while (!totals.has(total)) {
    totals.add(total);
    total += parseInt(inpArr[i++ % inpArr.length]);
}

console.log(`Day 1 | Part 2 - Solution: ${total}`);